src/foam/nanos/ruler/RuleEngine.java

```diff
@@ -196,7 +197,13 @@ private void applyAsyncRule(Rule rule, FObject obj, FObject oldObj) {
       if ( ! rule.f(x, nu, oldObj) ) return;
 
       PM pm = PM.create(getX(), RulerDAO.getOwnClassInfo(), "ASYNC", rule.getDaoKey(), rule.getId());
+      var before = nu.fclone();
```

Comment:
Yes, can you skip this if Operation.REMOVE.

Reply:
oh i see